clamav.py

```diff
@@ -195,21 +197,48 @@ def scan_file(path):
         stdout=subprocess.PIPE,
         env=av_env,
     )
-    output = av_proc.communicate()[0].decode()
-    print("clamdscan output:\n%s" % output)
 
-    # Turn the output into a data source we can read
-    summary = scan_output_to_json(output)
+    try:
+        output, errors = av_proc.communicate(timeout=CLAMDSCAN_TIMEOUT)
+    except subprocess.TimeoutExpired:
+        av_proc.kill()
+        output, errors = av_proc.communicate()
+
+    decoded_output = output.decode()
+    print("clamdscan output:\n%s" % decoded_output)
+
     if av_proc.returncode == 0:
         return AV_STATUS_CLEAN, AV_SIGNATURE_OK
     elif av_proc.returncode == 1:
+        # Turn the output into a data source we can read
+        summary = scan_output_to_json(decoded_output)
         signature = summary.get(path, AV_SIGNATURE_UNKNOWN)
         return AV_STATUS_INFECTED, signature
     else:
         msg = "Unexpected exit code from clamdscan: %s.\n" % av_proc.returncode
+
+        if errors:
+            msg += "Errors: %s\n" % errors.decode()
+
         print(msg)
         raise Exception(msg)
 
+def is_clamd_running():
+    clamd_socket = '/tmp/clamd.sock'
+    print("Checking if clamd is running on %s" %clamd_socket)
+
+    if os.path.exists(clamd_socket):
+        with socket.socket(socket.AF_UNIX, socket.SOCK_STREAM) as s:
```

Comment:
Good to see you're using a context manager here, so we don't need to manually ensure we close the socket 👍